conans/client/tools/scm.py

```diff
@@ -120,3 +125,79 @@ def get_revision(self):
             return commit
         except Exception as e:
             raise ConanException("Unable to get git commit from %s\n%s" % (self.folder, str(e)))
+
+
+class SVN(SCMBase):
+    cmd_command = "svn"
+
+    def __init__(self, runner=None, *args, **kwargs):
+        def runner_no_strip(command):
+            return subprocess.check_output(command, shell=True).decode()
+        runner = runner or runner_no_strip
+        super(SVN, self).__init__(runner=runner, *args, **kwargs)
+
+    def clone(self, url, branch=None, submodule=None):
+        assert branch is None, "This concept has no meaning for SVN"
+        assert submodule is None, "Here we can implement something about svn:externals"
+        assert os.path.exists(self.folder), "It guaranteed to exists according to SCMBase::__init__"
+
+        params = "--no-auth-cache --non-interactive"
+        params += " --trust-server-cert-failures=unknown-ca" if not self._verify_ssl else ""
+
+        if not os.listdir(self.folder):
+            url = self.get_url_with_credentials(url)
+            command = "co {url} . {params}".format(url=url, params=params)
+            output = self.run(command)
+        else:
+            output = self.run("revert . --recursive {params}".format(params=params))
+            output += self.run("cleanup . --remove-unversioned --remove-ignored {params}".format(params=params))
+
+        return output
+
+    def checkout(self, element):
+        # Element can only be a revision number
+        return self.run("update -r {rev}".format(rev=element))
+
+    def excluded_files(self):
+        excluded_list = []
+        output = self.run("status --no-ignore")
+        for it in output.splitlines():
+            if it[0] == 'I':
+                file = it[9:].strip()
+                excluded_list.append(os.path.normpath(os.path.join(self.folder, file)))
+        excluded_list.append(os.path.normpath(os.path.join(self.folder, ".svn")))
+        return excluded_list
+
+    def get_remote_url(self, remote_name=None):
+        assert remote_name is None, "This concept has no meaning for SVN"
+        return self.run("info --show-item url").strip()
+
+    def get_revision(self):
+        wc_revision = self.run("info --show-item revision").strip()
+
+        # Check if working copy is consistent
```

Comment:
Good point and a nice to have. But I'm not especially worried about it, we could start supposing that the user is following some good practices like continuous integration automation where the repository will be consistent.

Reply:
Good question. Here we don't have any output. In other similar situations we are passing an optional `output` to the constructor and then:

`out = out or ConanOutput(sys.stdout, True)`

The  `out` usage is the same as in the conanfile.output: `out.warn("C is not C++")`